go/bytes2/buffer.go

```diff
@@ -65,7 +65,7 @@ func (buf *Buffer) String() string {
 // is _not_ allocated, so modifying this buffer after calling StringUnsafe will lead
 // to undefined behavior.
 func (buf *Buffer) StringUnsafe() string {
-	return *(*string)(unsafe.Pointer(&buf.bytes))
+	return unsafe.String(unsafe.SliceData(buf.bytes), len(buf.bytes))
```

Comment:
I can pull this out into a separate change, but just came across it while doing all of this, and figured it'd be good to adopt the newer convention from go1.20.